src/ResourceManager/Media/Commands.Media/MediaService/SetAzureRmMediaService.cs

```diff
@@ -0,0 +1,146 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Linq;
+using System.Management.Automation;
+using System.Net;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.Commands.Media.Common;
+using Microsoft.Azure.Commands.Media.Models;
+using Microsoft.Azure.Management.Media;
+using Microsoft.Azure.Management.Media.Models;
+using RestMediaService = Microsoft.Azure.Management.Media.Models.MediaService;
+
+namespace Microsoft.Azure.Commands.Media.MediaService
+{
+    /// <summary>
+    /// Update a media service.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, MediaServiceNounStr, SupportsShouldProcess = true), OutputType(typeof(PSMediaService))]
+    public class SetAzureRmMediaService : AzureMediaServiceCmdletBase
+    {
+        private const string SetMediaServiceWarning = "Are you sure you want to set MediaService {0} ?";
+        private const string SetMediaServiceWhatIfMessage = "Set MediaService ";
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The media service account name.")]
+        [ValidateNotNullOrEmpty]
+        [ValidateLength(MediaServiceAccountNameMinLength, MediaServiceAccountNameMaxLength)]
+        [ValidatePattern(MediaServiceAccountNamePattern, Options = RegexOptions.None)]
+        [Alias(AccountNameAlias)]
+        public string AccountName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The tags associated with the media account.")]
+        [ValidateNotNull]
+        public Hashtable Tags { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The storage accounts assosiated with the media account.")]
+        [ValidateNotNull]
+        public PSStorageAccount[] StorageAccounts { get; set; }
+
+        [Parameter(Mandatory = false,
+           HelpMessage = "Force to set media service without confirm.")]
+        public SwitchParameter Force { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            var mediaServiceParams = new RestMediaService();
+
+            if (Tags != null)
```

Comment:
If (SHouldProcess) should really go here, to avoid any exceptions when a user specifies -WhatIf